core_java/twitter/src/test/java/ca/jrvs/apps/twitter/controller/TwitterControllerUnitTest.java

```diff
@@ -0,0 +1,90 @@
+package ca.jrvs.apps.twitter.controller;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import ca.jrvs.apps.twitter.model.Coordinates;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.service.Service;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TwitterControllerUnitTest {
+
+  @Mock
+  Service mockService;
+
+  @InjectMocks
+  TwitterController controller;
+
+  Tweet tweet, returnedTweet;
+
+  @Before
+  public void setUp() {
+    tweet = new Tweet("", 1, "1", "create tweet test #random", null, (new Coordinates(1, 2)), 0, 0,
+        false, false);
+  }
+  @Test
+
+  public void postTweet() {
+    doThrow(new IllegalArgumentException()).when(mockService).postTweet(any());
+    try {
+      controller.postTweet(new String[]{"text", "45:65"});
+    } catch (IllegalArgumentException e) {
+      assertTrue(true);
+    }
+
+    doReturn(tweet).when(mockService).postTweet(any());
+    returnedTweet = controller.postTweet(new String[]{"text", "45:12"});
+
+    assertEquals(tweet.getId(), returnedTweet.getId());
+    assertEquals(tweet.getText(), returnedTweet.getText());
+    assertTrue(tweet == returnedTweet);
+
+  }
+
+  @Test
+  public void showTweet() {
+    try {
+      controller.showTweet(new String[]{"12876sdhjgbjksaabcc"});
+    } catch (IllegalArgumentException e) {
+      assertTrue(true);
+    }
+
+    doReturn(tweet).when(mockService).showTweet("456", new String[0]);
+    returnedTweet = controller.showTweet(new String[]{"456"});
+
+    assertEquals(tweet.getId(), returnedTweet.getId());
+    assertEquals(tweet.getText(), returnedTweet.getText());
+    assertTrue(tweet == returnedTweet);
+  }
+
+  @Test
+  public void deleteTweet() {
+    try {
+      controller.deleteTweet(new String[]{"12876sdhjgbjksaabcc"});
+    } catch (IllegalArgumentException e) {
+      assertTrue(true);
+    }
+
+//    doReturn(Arrays.asList(tweet)).when(mockService).deleteTweets(new String[]{anyString()});
```

Comment:
This returns an empty Tweet list; I also tried with "when(...).thenReturn(...)" and initializing the list outside of the Mockito statement but got the same empty list; I stepped through the code and somewhere along the process a "StubbingException" is thrown because the "toString()" method of the stub is not found; I don't understand why such an exception is being thrown since I am only stubbing one thing (mockService). Am I missing something obvious?